projects/ui-components/lf-user-feedback/feedback-image-upload/feedback-image-upload.component.ts

```diff
@@ -66,53 +66,66 @@ export class FeedbackImageUploadComponent {
   }
 
   private async tryReadAndValidateImageAsync(image: File | undefined): Promise<boolean> {
-    if (!image) {
-      return false;
-    }
     try {
-      const isImageSupported = this.supportedImageTypeArray.includes(image.type);
-      if (!isImageSupported) {
-        throw new ImageUploadError(ImageUploadErrorType.UnsupportedFormat);
-      }
-      if (image.size <= this.imageSizeLimitBytes) {
-        const encodingData = await this.getBase64Async(image);
-        this.feedbackImageBase64.emit(encodingData);
-        this.imageUploaded = {
-          name: image.name, rawBase64: encodingData
-        };
-        return true;
-      } else {
-        throw new ImageUploadError(ImageUploadErrorType.TooLarge);
-      }
+      this.showLoader = true;
+      await this.uploadImageOrThrow(image);
     } catch (error: any) {
-      let errorMessage: string;
-      if (error.name === ImageUploadError_name) {
-        switch ((<ImageUploadError>error).imageUploadErrorType) {
-          case ImageUploadErrorType.TooLarge:
-            errorMessage = this.localizationService.getResourceStringComponents('IMAGE_EXCEEDS_MAX_FILE_SIZE_0', [`${this.megabyteLimit} MB`]);
-            break;
-          case ImageUploadErrorType.UnsupportedFormat:
-            errorMessage =
-              this.localizationService.getResourceStringComponents('IMAGE_CORRUPTED_UNRECOGNIZED_FORMAT') +
-              ' ' +
-              this.localizationService.getResourceStringComponents('ACCEPTED_FORMATS_ARE_0', [
-                this.acceptedImageFormats,
-              ]);
-            break;
-          default:
-            errorMessage = error.message ?? this.localizedStrings.UNKNOWN_ERROR;
-            break;
-        }
-      } else {
-        errorMessage = error.message ?? this.localizedStrings.UNKNOWN_ERROR;
+      this.handleImageUploadError(error);
+    } finally {
+      this.showLoader = false;
+      return !!this.imageUploaded;
+    }
+  }
+
+  private async uploadImageOrThrow(image: File | undefined) {
+    const imageValidationError = this.validateImage(image);
+    if (imageValidationError) {
+      throw imageValidationError;
+    } else {
+      const encodingData = await this.getBase64Async(image as File);
+      this.feedbackImageBase64.emit(encodingData);
+      this.imageUploaded = {
+        name: (<File>image).name, rawBase64: encodingData
+      };
+    }
+  }
+
+  private validateImage(image: File | undefined): Error | undefined{
+    if (!image) {
+      return new Error("image does not exist");
```

Comment:
instead of returning an error, throws the error and it will bubble up if the try/catch block of tryReadAndValidateImageAsync. This method can be return void.